microprofile/WFLY-14987_reactive_messaging-ssl-context-kafka-connector.adoc

```diff
@@ -0,0 +1,278 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14987[WFLY-14987]
+* https://issues.redhat.com/browse/EAP7-1664
+
+=== Related Issues
+
+
+=== Dev Contacts
+
+* mailto:{email}}[{author}]
+
+=== QE Contacts
+
+* mailto:jstourac@redhat.com[Jan Stourac]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+* WildFly - the Reactive Messaging integration will be enhanced to provide the functionality for this to work
+
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+* It must be possible to use the global connector or individual method format properties to specify an Elytron provided `SSLContext` (set up via the `/subsystem=elytron/client-ssl-context` resources and its dependencies).
+** If there is no Elytron provided `SSLContext` with that name an error will be thrown on deployment
+* The examples shown above for using properties from a deployment `microprofile-config.properties`. However, the configuration is using MicroProfile Config, so other mechanisms to specify these values, such as environment variables, are also valid.
+* The indicated Elytron `SSLContext` must be present at the time of deployment, or an error will be thrown.
+* There will be a dependency from the deployment on the JBoss MSC service providing the `SSLContext`.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+* Although we encourage users to use the Elytron provided `SSLContext` mechanism, it is not an error to use the native SmallRye Reactive Messaging/Apache Kafka client properties
+* If the user has specified the `<PREFIX>wildfly.elytron.ssl.context` property in their MicroProfile Config, it is not an error to use the other Kafka native ssl properties (e.g. the ones that start with `<PREFIX>.ssl.` in the configuration). These properties will simply be ignored.
+
+== Test Plan
+At the moment, it looks like the tests for this will need to reside in QE's testsuite.
+
+The issue is that we are currently using the Spring embedded Kafka test utility for testing the Kafka integration which does not have the required hooks. testcontainers.org is also popular but requires users have a running version of Docker on their machines, which is not the case for our CI.
```

Comment:
Hi @kabir, btw - reasons for not introducing testcontainers.org into WF TS due to lack of `docker` in CI is being broken by this PR https://github.com/wildfly/wildfly/pull/14604/files#diff-f1df13354db56727fb043478c855f938dcf79cd9b288c542c4b0262399d52cdeR76. So we shall probably reevaluate this reasoning :)

Reply:
@jstourac although the test added by the PR fails :) I presume that is the one needing test containers, but haven't checked :)